Lib/test/test_logging.py

```diff
@@ -4853,6 +4887,30 @@ def dummy_handle_error(record):
             # didn't write anything due to the encoding error
             self.assertEqual(data, r'')
 
+    def test_log_taskName(self):
+        async def log_record():
+            logging.warning('hello world')
+
+        try:
+            encoding = 'utf-8'
+            logging.basicConfig(filename='test.log', errors='strict', encoding=encoding,
+                                format='%(taskName)s - %(message)s', level=logging.WARNING)
+
+            self.assertEqual(len(logging.root.handlers), 1)
+            handler = logging.root.handlers[0]
+            self.assertIsInstance(handler, logging.FileHandler)
+
+            with asyncio.Runner(debug=True) as runner:
+                logging.logAsyncioTasks = True
+                runner.run(log_record())
+        finally:
+            asyncio.set_event_loop_policy(None)
+            handler.close()
+            with open('test.log', encoding='utf-8') as f:
+                data = f.read().strip()
+            os.remove('test.log')
+            self.assertRegex(data, r'Task-\d+ - hello world')
```

Comment:
```python
handler = None
try:
    ...
finally:
    asyncio.set_event_loop_policy(None)
    if handler is not None:
        handler.close()
    if os.path.exists('test.log'):
        ....
```

to prevent sub-exceptions in the finally block if there is an exception/assertion error in the try block ?

Reply:
I would have put the `self.assertRegex(data, r'Task-\d+ - hello world')` at the end of the try block too.

Reply:
Yes, that's a fair point. This test probably needs tidying up a little.

Reply:
I was following the pattern of the encoding tests above this one.
Happy to raise a new PR and fix it up along with the others.

Reply:
That's OK, I've just done it. I'm working on logging tests for something else.

Reply:
It's #93265.